src/prommis/solvent_extraction/ree_aq_distribution.py

```diff
@@ -158,5 +158,17 @@ def get_material_flow_terms(self, p, j):
                 to_units=units.mol / units.hour,
             )
 
+    def get_material_density_terms(self, p, j):
+        if j == "H2SO4":
```

Comment:
Why are you using H2SO4 here (and I note that earlier you refer to this as the solvent). This presumes that the concentration of H2SO4 is effectively constant (i.e. no dissociation) which is not correct.

The approach you are trying to use here works well-enough for the concentration of H2O in dilute aqueous solutions (where the concentration of H2O is essentially constant), but I do not think it is appropriate here (H2SO4 is not constant or the dominant species).

Reply:
No, I am not suggesting anything to do with dissociation. I am pointing out that you have made two assumptions here that are questionable, if not incorrect.

Firstly, the purpose of this method here is to determine how the accumulation term for each species will be calculated; i.e. this is effectively the concentration of each species. The reason we do it this way it to allow for cases where we do not have a concentration variable directly available. Thus, what you have effectively done here is say that the concentration of H2SO4 will be constant in the system and cannot change, which is almost definitely incorrect.

Now, you probably did this by copying other code where I had done this, but I suspect you missed a key point in those examples. They were all cases where we had dilute aqueous solutions, and this was only applied to the H2O in the system. For dilute aqueous solutions, we can assume that the concentration of the water dominates the system and is thus constant. However, this is only true when you consider a dilute system with a single bulk solvent.

Looking at your comments, you have stated that you are considering H2SO4 to be the solvent in this system, which is incorrect. The solvent is H2O still, and H2SO4 is a dissolved solute (which will likely vary in concentration). Also, unlike dilute aqueous systems, H2SO4 solutions also shows a distinct variation in density with concentration, thus the assumption of constant density is questionable (although probably OK for narrow ranges, which is what I did in the leaching properties).

Basically, you need to rethink how you are defining the system, and probably switch to what the leaching properties did.

Reply:
Okay I get your point, so the solvent is H2O. I have to think about the H2SO4 concentration part

Reply:
Not really, you just need to realize that H2SO4 is a dissolved species and not H2O (line 49). Then, look at how many dissolved species you have versus how many total species, and think about why that is the case and what it means for writing the material balances.

The issue you have here is that you are copying something I did where I took some shortcuts without understanding what these shortcuts were or why I took them. You might want to step back and think about how you would write this yourself (in terms of equations, not code) first.

Reply:
For now, I have switched to leaching property package for steady dynamics, and it is working fine. That package uses water as dominant component, so I think this is fine.